presto-native-execution/presto_cpp/main/SystemConnector.cpp

```diff
@@ -0,0 +1,251 @@
+  auto numRows = taskMap.size();
+  taskTableResult->resize(numRows);
+  for (auto i = 0; i < taskTableResult->type()->size(); i++) {
+    taskTableResult->childAt(i)->resize(numRows);
+  }
+
+  auto toMillis = [](int64_t nanos) -> int64_t { return nanos / 1'000'000; };
+  int i = 0;
+  for (const auto& taskEntry : taskMap) {
+    auto task = taskEntry.second;
+    auto taskInfo = task->updateInfo();
+    taskTableResult->childAt(0)->asFlatVector<StringView>()->set(
+        i, StringView(taskInfo.nodeId));
+    taskTableResult->childAt(1)->asFlatVector<StringView>()->set(
+        i, StringView(taskInfo.taskId));
+    std::string temp = fmt::format("{}", task->id.stageExecutionId());
+    taskTableResult->childAt(2)->asFlatVector<StringView>()->set(
+        i, StringView(temp));
+    temp = fmt::format("{}", task->id.stageId());
+    taskTableResult->childAt(3)->asFlatVector<StringView>()->set(
+        i, StringView(temp));
+    temp = fmt::format("{}", task->id.queryId());
+    taskTableResult->childAt(4)->asFlatVector<StringView>()->set(
+        i, StringView(temp));
+    temp = fmt::format("{}", taskInfo.taskStatus.state);
+    taskTableResult->childAt(5)->asFlatVector<StringView>()->set(
+        i, StringView(temp));
+    taskTableResult->childAt(6)->asFlatVector<int64_t>()->set(
+        i, taskInfo.stats.totalDrivers);
+    taskTableResult->childAt(7)->asFlatVector<int64_t>()->set(
+        i, taskInfo.stats.queuedDrivers);
+    taskTableResult->childAt(8)->asFlatVector<int64_t>()->set(
+        i, taskInfo.stats.runningDrivers);
+    taskTableResult->childAt(9)->asFlatVector<int64_t>()->set(
+        i, taskInfo.stats.completedDrivers);
+    taskTableResult->childAt(10)->asFlatVector<int64_t>()->set(
+        i, toMillis(taskInfo.stats.totalScheduledTimeInNanos));
+    taskTableResult->childAt(11)->asFlatVector<int64_t>()->set(
+        i, toMillis(taskInfo.stats.totalCpuTimeInNanos));
+    taskTableResult->childAt(12)->asFlatVector<int64_t>()->set(
+        i, toMillis(taskInfo.stats.totalBlockedTimeInNanos));
+    taskTableResult->childAt(13)->asFlatVector<int64_t>()->set(
+        i, taskInfo.stats.rawInputDataSizeInBytes);
+    taskTableResult->childAt(14)->asFlatVector<int64_t>()->set(
+        i, taskInfo.stats.rawInputPositions);
+    taskTableResult->childAt(15)->asFlatVector<int64_t>()->set(
+        i, taskInfo.stats.processedInputDataSizeInBytes);
+    taskTableResult->childAt(16)->asFlatVector<int64_t>()->set(
+        i, taskInfo.stats.processedInputPositions);
+    taskTableResult->childAt(17)->asFlatVector<int64_t>()->set(
+        i, taskInfo.stats.outputDataSizeInBytes);
+    taskTableResult->childAt(18)->asFlatVector<int64_t>()->set(
+        i, taskInfo.stats.outputPositions);
+    taskTableResult->childAt(19)->asFlatVector<int64_t>()->set(
+        i, taskInfo.stats.physicalWrittenDataSizeInBytes);
+    // Setting empty TIMESTAMP for below fields as string -> Timestamp is not
```

Comment:
This is hacky. Velox reports timestamps as integers, but then we use PrestoTask::updateInfoLocked to convert these to strings, and then complain here that we cannot convert them back to integers. Please, consider skipping unnecessary conversions.

```
  prestoTaskStats.createTime =
      util::toISOTimestamp(taskStats.executionStartTimeMs);
  prestoTaskStats.firstStartTime =
      util::toISOTimestamp(taskStats.firstSplitStartTimeMs);
  prestoTaskStats.lastStartTime =
      util::toISOTimestamp(taskStats.lastSplitStartTimeMs);
  prestoTaskStats.lastEndTime =
      util::toISOTimestamp(taskStats.executionEndTimeMs);

  /// Epoch time (ms) when task starts to run
  uint64_t executionStartTimeMs{0};

```

Reply:
@aditi-pandit What do you think?

Reply:
Am tracking these separately in the PrestoTask now.